src/hotspot/share/runtime/threads.cpp

```diff
@@ -1430,6 +1430,7 @@ JavaThread* Threads::owning_thread_from_object(ThreadsList * t_list, oop obj) {
 }
 
 JavaThread* Threads::owning_thread_from_monitor(ThreadsList* t_list, ObjectMonitor* monitor) {
+  assert(SafepointSynchronize::is_at_safepoint(), "not safe outside of safepoint");
```

Comment:
Shouldn't this be gated on UseFastLocking?

Reply:
Hmmm.... `owning_thread_from_monitor()` is only called from
`ObjectSynchronizer::get_lock_owner()` when `get_lock_owner()`
knows that it has an ObjectMonitor in hand. I'm not at all sure that
we can assert that `ObjectSynchronizer::get_lock_owner()` is
only called from a safepoint.

Reply:
There has been a single call path from management.cpp that is not calling this code at a safepoint, and I changed that code to take the safepoint-taking variant code-path when using UseFastLocking. That path already existed and has been used when max_depth == 0.